src/main/groovy/com/github/winplay02/gitcraft/meta/VersionInfo.java

```diff
@@ -0,0 +1,16 @@
+package com.github.winplay02.gitcraft.meta;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+public record VersionInfo(ArtifactMetadata assetIndex, String assets, Downloads downloads, String id,
+						  JavaVersion javaVersion, List<LibraryMetadata> libraries, String mainClass,
+						  ZonedDateTime releaseTime, ZonedDateTime time, String type) {
+	public record Downloads(ArtifactMetadata client, ArtifactMetadata client_mappings, ArtifactMetadata server,
+									   ArtifactMetadata server_mappings, ArtifactMetadata windows_server, ArtifactMetadata server_zip) {
+	}
+
```

Comment:
Could you expand on that? It makes me wonder what kind of data is missing from the "main" metadata json that is present in (one of) the jsons listed in the details json.

Reply:
This was mostly about Mojang switching the assets index of already released snapshots, as soon as new snapshots / versions are released. These are only available in a version manifest, and not available in the version details.

The other thing, the merge functionality allows, is to handle version manifests that only contain server / only contain clients and link them together (only within one version). But this information is available in the details, so that would not be needed anymore.

But as said, this is not needed right now and still needs some thoughts, so it will return later, when needed.